File: src/main/java/org/kobe/xbot/Utilites/Utilities.java
package org.kobe.xbot.Utilites;

public class Utilities {
    public static boolean validateKey(String key) {
        // Check if key is null or empty
        if (key == null) {
            throw new IllegalArgumentException("Key cannot be null.");
        }

        // Check if key contains spaces
        if (key.contains(" ")) {
            throw new IllegalArgumentException("Key cannot contain spaces.");
        }

        // Check if key starts or ends with '.'
        if (key.startsWith(".") || key.endsWith(".")) {
            throw new IllegalArgumentException("Key cannot start or end with '.'");
        }

        // Check if key contains multiple consecutive '.'
        if (key.contains("..")) {
            throw new IllegalArgumentException("Key cannot contain multiple consecutive '.'");
        }

        // Check if each part of the key separated by '.' is empty
        if (!key.isEmpty()) {
            String[] parts = key.split("\\.");
            for (String part : parts) {
                if (part.isEmpty()) {
                    throw new IllegalArgumentException("Key contains empty part(s).");
                }
            }
        }
        return true;
    }

    public static boolean validateName(String name, boolean throwError) {
        // Check if name is null or empty
        if (name == null) {
            if (throwError) throw new IllegalArgumentException("Name cannot be null.");
            else return false;
        }

        // Check if name contains spaces
        if (name.contains(" ")) {
            if (throwError) throw new IllegalArgumentException("Name cannot contain spaces.");
            else return false;
        }

        // Check if name contains '.'
        if (name.contains(".")) {
            if (throwError) throw new IllegalArgumentException("Name cannot contain '.'");
            else return false;
        }


        return true;
    }
}

File: src/main/python/XTABLES/Utilities.py
def validate_key(key, throw_error):
    # Check if the key is null or empty
    if key is None:
        if throw_error:
            raise ValueError("Key cannot be null.")
        else:
            return False

    # Check if key contains spaces
    if " " in key:
        if throw_error:
            raise ValueError("Key cannot contain spaces.")
        else:
            return False

    # Check if key starts or ends with '.'
    if key.startswith(".") or key.endswith("."):
        if throw_error:
            raise ValueError("Key cannot start or end with '.'")
        else:
            return False

    # Check if key contains multiple consecutive '.'
    if ".." in key:
        if throw_error:
            raise ValueError("Key cannot contain multiple consecutive '.'")
        else:
            return False

    # Check if each part of the key separated by '.' is empty
    if key:
        parts = key.split(".")
        for part in parts:
            if not part:
                if throw_error:
                    raise ValueError("Key contains empty part(s).")
                else:
                    return False

    return True


def parse_string(s):
    # Check if the string starts and ends with \" and remove them
    if s.startswith('"\\"') and s.endswith('\\""'):
        s = s[3:-3]
    elif s.startswith('"') and s.endswith('"'):
        s = s[1:-1]
    return s


def validate_name(name, throw_error):
    # Check if the name is null or empty
    if name is None:
        if throw_error:
            raise ValueError("Name cannot be null.")
        else:
            return False

    # Check if the name contains spaces
    if " " in name:
        if throw_error:
            raise ValueError("Name cannot contain spaces.")
        else:
            return False

    # Check if the name contains '.'
    if "." in name:
        if throw_error:
            raise ValueError("Name cannot contain '.'")
        else:
            return False

    return True
